wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -80,6 +80,30 @@ public static function collectionCampTabset($tabsetName, &$tabs, $context) {
       return;
     }
 
+    // Get the current logged-in user's contact ID and permissions.
+    $currentUserId = \CRM_Core_Session::getLoggedInContactID();
+    if (!$currentUserId) {
+      \Civi::log()->error('No CiviCRM user context available.');
+      return;
+    }
```

Comment:
What is the use of this variable `$currentUserId`? It looks like `\CRM_Core_Permission::check` is enough. Isn't it?